File: php/strategy/MallardDuck.php
<?
namespace strategy;

class MallardDuck extends Duck {
    public function __construct() {
        parent::__construct();
        $this->flyBehavior = new FlyWithWings();
        $this->quackBehavior = new Quack();
    }

    public function display() {
        echo "I'm a real Mallard duck";
    }
}
?>

File: python/2.7/strategy/MallardDuck.py
from FlyWithWings import *
from Quack import *

from Duck import *

class MallardDuck(Duck):
	def __init__(self):
		super(MallardDuck, self).__init__()
		self.flyBehavior = FlyWithWings()
		self.quackBehavior = Quack()

	def display(self):
		print "I'm a real Mallard duck"
